products/bash/setup-demos.sh

```diff
@@ -155,16 +162,58 @@ function merge_addon {
   echo ${ADDON_ARRAY_JSON}
 }
 
+function update_status() {
+  MESSAGE=${1}
+  TYPE=${2}        # resource type
+  RESULT_CODE=${3} # success(0), error(1)
+  TIMESTAMP=${4}
+  REASON=${5}        # command type
+  PHASE=${6}         # Pending, Running or Failed
+  RESOURCE_NAME=${7} # name of the resource
+  CONDITION_TYPE=""  # for the type in conditions
+
+  $DEBUG && divider && echo -e "$info update_status(): message($MESSAGE) - type($TYPE) - resultcode($RESULT_CODE) - timestamp($TIMESTAMP) - reason($REASON) - phase($PHASE) - resourceName($RESOURCE_NAME)"
+
+  if [[ $RESULT_CODE -eq 1 ]]; then
+    CONDITION_TYPE="Error"
+  elif [[ $RESULT_CODE -eq 0 ]]; then
+    CONDITION_TYPE="Success"
+  else
+    CONDITION_TYPE="Pending"
+  fi
+
+  # update condition array if error occurred
+  CONDITION_TO_ADD=$(echo $CONDITION_ELEMENT_OBJECT | jq -r '.message="'"$TYPE - $MESSAGE"'" | .status="True" | .type="'$CONDITION_TYPE'" | .lastTransitionTime="'$TIMESTAMP'" | .reason="'$REASON'" ')
+  # add condition to condition array
+  STATUS=$(echo $STATUS | jq -c '.conditions += ['"${CONDITION_TO_ADD}"']')
+
+  if [[ ("$TYPE" == "namespace") && ("$REASON" == "Getting" || "$REASON" == "Creating") && ($RESULT_CODE -eq 0) ]]; then
```

Comment:
I'm not keen on this, it seems like it makes this function not general purpose, it will need to be updated to handle all of the different failure types. I guess this function can be simpler if (as per comment elsewhere) we only add conditions for abnormal conditions.